pkg/agent/agent.go

```diff
@@ -158,6 +159,15 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 		}
 	}
 
+	for _, pod := range pods {
+		glog.Infof("Waiting for pod %q to terminate", pod.Name)
+		if err := k.waitForPodDeletion(pod); err != nil {
+			glog.Errorf("failed waiting for pod %q to terminate: %v", pod.Name, err)
```

Comment:
In practice, many of these log lines will be about transient errors to make API calls and we shouldn't stop the wait for them (e.g. "failed waiting for pod xx to terminate: can't reach 10.3.0.1"). Other uses of wait in agent allow the anon function to log errors and return nil to continue trying.

Reply:
On timeout, we can say we're skipping waiting for the pod to terminate.